ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -20,5 +39,240 @@ public override string Name
+                    }
+                }
+            }
+            callback.Invoke(result);
+        }
+
+        /// <summary>
+        /// Inserts multiple (key, value) pairs. The insertion will replace conflicting (key, value) pairs.
+        /// </summary>
+        /// <param name="keyValueArray">Array of key values</param>
+        /// <param name="callback">Callback.</param>
+        [ReactMethod]
+        public void multiSet(JArray keyValueArray, ICallback callback)
+        {
+            if (keyValueArray == null)
+                throw new ArgumentNullException(nameof(keyValueArray));
+
+            var result = new JArray();
+
+            foreach (var keyValue in keyValueArray)
+            {
+                if (keyValue.Type == JTokenType.Array && keyValue.ToObject<JArray>().Count == 2)
+                {
+                    var pair = keyValue.ToObject<JArray>();
+                    if (pair.First.Type == JTokenType.String)
+                    {
+                        var key = pair.First.ToObject<string>();
+                        switch (pair.Last.Type)
+                        {
+                            case JTokenType.Null:
+                                _container.Values[key] = _nul;
+                                break;
+                            case JTokenType.Boolean:
+                                _container.Values[key] = pair.Last.ToObject<bool>();
+                                break;
+                            case JTokenType.Integer:
+                                _container.Values[key] = pair.Last.ToObject<long>();
+                                break;
+                            case JTokenType.Float:
+                                _container.Values[key] = pair.Last.ToObject<double>();
+                                break;
+                            case JTokenType.String:
+                                _container.Values[key] = _str + pair.Last.ToObject<string>();
+                                break;
+                            case JTokenType.Array:
+                                _container.Values[key] = _arr + pair.Last.ToString();
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        result.Add(new JArray { _invalidKey, pair.First });
+                    }
+                }
+                else
+                {
+                    result.Add(new JArray { _invalidPair, keyValue });
+                }
+            }
+            callback?.Invoke(result);
+        }
+
+        /// <summary>
+        /// Removes all rows of the keys given.
+        /// </summary>
+        /// <param name="keys">Array of key values</param>
+        /// <param name="callback">Callback.</param>
+        [ReactMethod]
+        public void multiRemove(JArray keys, ICallback callback)
```

Comment:
Honestly, I would just use string[] here.  We use Newtonsoft.Json parsing when we bind parameters to native methods, so binding to a string[] should be fine here.